modules/assim.sequential/R/sda.enkf.R

```diff
@@ -83,19 +96,11 @@ sda.enkf <- function(settings, IC, prior, obs.mean, obs.cov, variables,
     print(paste("please make sure that the PEcAn interface is loaded for",model))
     stop()
   }
-  
-  ## split clim file
-  ## leaving in for sipnet
-  #   full.met <- settings$run$inputs$met$path
-  #   new.met  <- file.path(settings$rundir,basename(full.met))
-  #   file.copy(full.met,new.met)
-  #   met <- new.met#split.met.SIPNET(new.met)
-  
-  
+ 
   ###-------------------------------------------------------------------###
   ### perform initial set of runs                                       ###
   ###-------------------------------------------------------------------###  
-  X = IC
+  X = matrix(NA,as.numeric(settings$ensemble$size),length(settings$pft)) #why does X need to be here?
```

Comment:
You've dropped the IC as an argument to the function without moving it into the settings or moving the IC generating function into sda.enkf. Then you set the IC matrix to all NA's, so I'm not sure how SIPNET is going to run if you give it NA for all its initial biomass pools.  Also, the dimension should be ensembles by state variables, not ensembles by PFTs